### PR TITLE
Improve test_merge_report

### PR DESCRIPTION
**Pull request by bruno-fs**
- further "pytestify" this test
- fix small errors
- improve the tests (now they actually test something)
- highlight some potential bugs that affect the current implementation.

### CODE REVIEW

File: qpc/report/test_report_merge.py [hunk 1]
```diff
@@ -73,295 +120,324 @@ def setup_class(cls):
...
-                    in report_out.getvalue().strip()
-                )
-
-    def test_detail_merge_error_json_files(self):
+            self.command.main(args)
+        # TODO: the following assertion is not being matched because the actual last
+        # logging line is an error when attempting to connect to quipucords. This is
+        # likely a bug because if qpc knows 'non_exist_file' doesn't exist, there's
+        # no report to merge
+        assert caplog.messages[-1] == messages.FILE_NOT_FOUND % non_exist_file
```
Comment (bruno-fs) [inline qpc/report/test_report_merge.py:247]: either a bug or a design flaw. didn't took the time to investigate

File: qpc/report/test_report_merge.py [hunk 2]
```diff
@@ -73,295 +120,324 @@ def setup_class(cls):
...
+            in caplog.text
+        )
+        assert (
+            messages.REPORT_JSON_MISSING_ATTR % {"file": bad_details2, "key": "facts"}
+            in caplog.text
+        )
+        assert (
+            messages.REPORT_JSON_MISSING_ATTR % {"file": bad_details3, "key": "facts"}
+            in caplog.text
+        )
```
Comment (bruno-fs) [inline qpc/report/test_report_merge.py:405]: the error message highlights that bad_details2 and 3 probably need to be redesigned

Comment (infinitewarp) [reply qpc/report/test_report_merge.py:405]: Add this as an inline comment for future readers? I appreciate the other inline comments you've added here around other weird behaviors.

**Review by infinitewarp — APPROVED** [review]
This feels like a great improvement, and I really appreciate that additional inline comments that call out weird or unexpected behaviors.

**Comment by abellotti** [comment]
Thank you @bruno-fs for taking care of this.

**Review by abellotti — APPROVED** [review]

**Pull request merged**